timrlink.net.CLI/Actions/ProjectTimeDatabaseExportAction.cs

```diff
@@ -57,36 +102,62 @@ public async Task Execute()
-                        StartTime = pt.startTime,
-                        EndTime = pt.endTime,
-                        Duration = pt.duration,
-                        BreakTime = pt.breakTime,
-                        Changed = pt.changed,
-                        Closed = pt.closed,
-                        StartPosition = LatLon(pt.startPosition),
-                        EndPosition = LatLon(pt.endPosition),
-                        LastModifiedTime = pt.lastModifiedTime,
-                        Task = JsonConvert.SerializeObject(BuildTaskPath(pt.taskUuid, taskDict).Select(task => task.name).ToArray()),
-                        Description = pt.description,
-                        Billable = pt.billable
-                    };
+                    var projectTime = context.ProjectTimes
+                        .Where(projectTime => projectTime.UUID.ToString() == pt.uuid)
+                        .FirstOrDefault() ?? new ProjectTime();
+
+                    projectTime.UUID = Guid.Parse(pt.uuid);
+                    projectTime.User = user != null ? $"{user.lastname} {user.firstname}" : pt.userUuid;
+                    projectTime.StartTime = pt.startTime;
+                    projectTime.EndTime = pt.endTime;
+                    projectTime.Duration = pt.duration;
+                    projectTime.BreakTime = pt.breakTime;
+                    projectTime.Changed = pt.changed;
+                    projectTime.Closed = pt.closed;
+                    projectTime.StartPosition = LatLon(pt.startPosition);
+                    projectTime.EndPosition = LatLon(pt.endPosition);
+                    projectTime.LastModifiedTime = pt.lastModifiedTime;
+                    projectTime.Task =
+                        JsonConvert.SerializeObject(BuildTaskPath(pt.taskUuid, taskDict).Select(task => task.name)
+                            .ToArray());
+                    projectTime.Description = pt.description;
+                    projectTime.Billable = pt.billable;
+                    projectTime.Deleted = null;
+
+                    return projectTime;
                 });
-
-                await context.ProjectTimes.AddOrUpdateRange(dbEntities);
-                await context.SaveChangesAsync();
+                
+                await context.ProjectTimes.AddOrUpdateRange(dbEntities);    
             }
+        
+            if (fromDate != null)
+            {
+                // Flag records that are not found anymore Deleted.
+                var projectTimesInDatabase = context.ProjectTimes.Where(projectTime =>
+                        projectTime.StartTime > fromDate && projectTime.EndTime < toDate)
+                    .ToList();
+                    
+                foreach (var projectTime in projectTimesInDatabase)
+                {
+                    if (!projectTimeUuids.Contains(projectTime.UUID))
+                    {
+                        projectTime.Deleted = DateTimeOffset.Now;
```

Comment:
we should use a single instance here, so timestamps don't shift for a single run - what about `importTime`?